src/styles/main.scss

```diff
@@ -1,3 +1,42 @@
 body {
   margin: 0;
 }
+
+.block {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 300px;
+  min-width: 300px;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 100px;
+  color: #fff;
+}
+
+$colors:
+  "100" rgb(100%, 0%, 0%),
+  "800" rgb(80%, 0%, 0%),
+  "600" rgb(60%, 0%, 0%),
+  "400" rgb(40%, 0%, 0%),
+  "200" rgb(20%, 0%, 0%),
+  "000" rgb(0%, 0%, 0%);
+
+@each $name, $color in $colors {
+  .block--color#{$name} {
+    background-color: $color;
+  }
+}
```

Comment:
You can also use $index and :nth-child(#{$index}) instead of $name and modificators)